src/OpenIddict.Security/OpenIddictExtensions.cs

```diff
@@ -0,0 +1,54 @@
+﻿using Microsoft.Extensions.Internal;
+using NWebsec.Middleware;
+using System;
+
+namespace Microsoft.AspNet.Builder {
+    public static class OpenIddictExtensions {
+        public static OpenIddictBuilder UseNWebsec([NotNull] this OpenIddictBuilder builder, Action<IFluentCspOptions> cspOptions) {
+
+            //Add security headers to the app.
+            return builder.AddModule("NWebsec", -20, app => {
+                // Insert a new middleware responsible of setting the Content-Security-Policy header.
+                // See https://nwebsec.codeplex.com/wikipage?title=Configuring%20Content%20Security%20Policy&referringTitle=NWebsec
+                app.UseCsp(options => {
```

Comment:
Consider using just here `app.UseCsp(cspOptions)` and moving the default directives to the other overload.